openedx/features/subscriptions/api/v1/views.py

```diff
@@ -0,0 +1,55 @@
+"""
+HTTP endpoints for interacting with subscriptions.
+"""
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.generics import ListAPIView, RetrieveUpdateAPIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework_oauth.authentication import OAuth2Authentication
+
+from django.http import Http404
+
+from lms.djangoapps.commerce.api.v1.permissions import ApiKeyOrModelPermission
+from openedx.core.lib.api.mixins import PutAsCreateMixin
+from openedx.features.subscriptions.models import UserSubscription
+from openedx.features.subscriptions.api.v1.serializers import UserSubscriptionSerializer
+
+
+class SubscriptionsListView(ListAPIView):
+    """
+    List subscriptions.
+    """
+    authentication_classes = (JwtAuthentication, OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (IsAuthenticated,)
+    serializer_class = UserSubscriptionSerializer
+
+    def get_queryset(self):
+        valid_subscriptions = True if self.request.query_params.get('valid') == 'true' else False
+        user_id = self.request.query_params.get('user', self.request.user.id)
+        if valid_subscriptions:
+            return UserSubscription.get_valid_subscription(user_id=user_id)
+
+        return UserSubscription.objects.filter(user_id=user_id)
+
+
+class SubscriptionRetrieveUpdateView(PutAsCreateMixin, RetrieveUpdateAPIView):
+    """
+    Retrieve, update, or create UserSubscriptions.
+    """
+    lookup_field = 'subscription_id'
+    lookup_url_kwarg = 'subscription_id'
+    model = UserSubscription
+    authentication_classes = (JwtAuthentication, OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (ApiKeyOrModelPermission,)
+    queryset = UserSubscription.objects.all()
+    serializer_class = UserSubscriptionSerializer
```

Comment:
I think we also need to add default order base of `created_at`
https://www.django-rest-framework.org/api-guide/filtering/#specifying-a-default-ordering

Reply:
done

Reply:
done